### ISSUE TITLE
channel shuffle

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#844]
# Reference
- [channel shuffle](https://zhuanlan.zhihu.com/p/125774455)
- [通道随机混合操作（Channel Shuffle Operation）](https://blog.csdn.net/u011501388/article/details/96322572)
- [用卷积实现 ShuffleNet 中 channel shuffle](https://zhuanlan.zhihu.com/p/203549964)

# Brief
- ShuffleNet 中引入了 channel shuffle, 用来进行不同分组的特征之间的信息流动, 以提高性能

```
import torch


def shuffle_channel(x, num_groups):
    batch_size, num_channels, height, width = x.size()
    assert num_channels % num_groups == 0

    x = x.view(batch_size, num_groups, num_channels // num_groups, height, width)
    x = x.permute(0, 2, 1, 3, 4)
    return x.contiguous().view(batch_size, num_channels, height, width)
```
